experimental/yarpl/include/yarpl/Flowables.h

```diff
@@ -44,23 +46,28 @@ class Flowables {
   }
 
   template <typename T>
-  static Reference<Flowable<T>> just(std::initializer_list<T> list) {
-    auto lambda = [ list, it = list.begin() ](
+  static Reference<Flowable<T>> justN(std::initializer_list<T> list) {
+    std::vector<T> vec;
+    std::move(list.begin(), list.end(), std::back_inserter(vec));
+
+    size_t i = 0;
+
+    auto lambda = [ v = std::move(vec), i ](
         Subscriber<T> & subscriber, int64_t requested) mutable {
       int64_t emitted = 0;
       bool done = false;
 
-      while (it != list.end() && emitted < requested) {
-        subscriber.onNext(*it++);
+      while (i < v.size() && emitted < requested) {
```

Comment:
I guess this is the same thing as the above, if its not used outside the lambda then you don't need to define it outside the lambda and capture it - you can just have it in the lambda's local scope.